### DISCUSSION TITLE
python-encoded Unicode are invalid?

### DISCUSSION THREAD

**Discussion by pgcd** [Kludex/uvicorn#1651] [Potential Issue] [answered]
I'm not entirely sure if the problem is actually in `uvicorn` or one of the layers "above" it (gunicorn and nginx), and it might be a simple configuration problem, so any advice is welcome.
We have (inherited) some less than ideal slugs, that look like "`searchterm-münchen`". I don't like them for a number of reasons but I'm not allowed to change them for SEO reasons. Still, they worked fine with `uwsgi`, and then with `gunicorn`; unfortunately, though, when I switched gunicorn workers to uvicorn, I started receiving errors like
`httptools.parser.errors.HttpParserInvalidURLError: invalid url b'/searchterm-m\xc3\xbcnchen' `
So, my question is: is this a code error and I should open an issue, or is it the expected behavior and I need to configure something to make it work?

**Comment by iudeen** [comment, marked as answer]
What version of Uvicorn are you using? 

Upgrade version to the latest. 

This issue was fixed in https://github.com/encode/uvicorn/pull/1263

**Reply by pgcd** [reply]
uvicorn[standard]==0.17.6

**Reply by iudeen** [reply]
Upgrade to the latest and check if you still have the issue.

I think this or something similar was fixed earlier. (Will add references later)

Edit:
This was fixed in https://github.com/encode/uvicorn/pull/1263

**Reply by pgcd** [reply]
I didn't try upgrading yet because I couldn't find anything clearly related in the issues, but worth a shot, thanks.

**Reply by pgcd** [reply]
And yes, looks like that did the trick, thank you!
